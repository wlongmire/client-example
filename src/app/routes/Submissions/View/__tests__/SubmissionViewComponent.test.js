/* eslint-env node, jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { SubmissionView } from '../index'

jest.mock('./../../../../actions/userActions', () => 'testActions')
jest.mock('./../../../../actions/submissionActions', () => 'testActions')

const initialState = {
  user: {
    _id: '5807af4bbd9f641100e26138',
    username: 'betamarshuser',
    firstName: 'Marsh',
    lastName: 'Beta',
    role: 'user',
    _brokerId: '5807af2edcba0f490c718471',
    accountPending: false
  },
  submissionData: [
    {
      _id: '59413679bafa390001ced695',
      primaryInsuredName: 'Andriy Kulak',
      totalCost: '$23,213,234',
      quotedPremium: '$21,612',
      totalPremium: '$22,693',
      type: 'ocp',
      updatedAt: '2017-06-14T13:13:29.776Z',
      dateCreated: '06-14-17 09:13am',
      dateUpdated: '06-14-17 09:13am',
      quoteStatus: 'Yes'
    }
  ]
}

// Snapshot for Home Header Component
describe('>>> SubmissionView --- Snapshot', () => {
  it('+++capturing Snapshot with', () => {
    const renderedValue = shallow(
      <SubmissionView
        submissions={initialState.submissionData}
        editSubmission={jest.fn()}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})

describe('>>> SubmissionView--- Shallow + passing the {store} directly', () => {
  let container

  beforeEach(() => {
    container = shallow(
      <SubmissionView
        submissions={[]}
        submissionData={initialState.submissionData}
        editSubmission={jest.fn()}
      />)
  })

  it('+++ check Prop matches with initialState', () => {
    const instance = container.instance()
    expect(instance.props.submissionData[0]._id).toEqual('59413679bafa390001ced695')
  })

  it('+++ check Prop matches with initialState', () => {
    const instance = container.instance()
    expect(instance.props.submissionData[0].primaryInsuredName).toEqual('Andriy Kulak')
  })
})
