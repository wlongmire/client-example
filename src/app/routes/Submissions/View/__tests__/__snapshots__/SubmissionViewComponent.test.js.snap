// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> SubmissionView --- Snapshot +++capturing Snapshot with 1`] = `
<div>
  <h3>
    Your Submissions
  </h3>
  <BootstrapTable
    bodyContainerClass={null}
    bordered={true}
    cellEdit={
      Object {
        "afterSaveCell": undefined,
        "beforeSaveCell": undefined,
        "blurToSave": false,
        "mode": "none",
        "nonEditableRows": undefined,
      }
    }
    columnFilter={false}
    condensed={true}
    containerClass={null}
    csvFileName="spreadsheet.csv"
    data={
      Array [
        Object {
          "_id": "59413679bafa390001ced695",
          "dateCreated": "06-14-17 09:13am",
          "dateUpdated": "06-14-17 09:13am",
          "primaryInsuredName": "Andriy Kulak",
          "quoteStatus": "Yes",
          "quotedPremium": "$21,612",
          "totalCost": "$23,213,234",
          "totalPremium": "$22,693",
          "type": "ocp",
          "updatedAt": "2017-06-14T13:13:29.776Z",
        },
      ]
    }
    deleteRow={false}
    exportCSV={false}
    fetchInfo={
      Object {
        "dataTotalSize": 0,
      }
    }
    headerContainerClass={null}
    height="100%"
    hover={false}
    ignoreSinglePage={false}
    insertRow={false}
    multiColumnSearch={true}
    multiColumnSort={1}
    options={
      Object {
        "defaultSortName": "updatedAt",
        "defaultSortOrder": "desc",
      }
    }
    pagination={true}
    printable={false}
    search={true}
    selectRow={
      Object {
        "bgColor": "",
        "clickToExpand": false,
        "clickToSelect": false,
        "clickToSelectAndEditCell": false,
        "customComponent": undefined,
        "hideSelectColumn": false,
        "mode": "none",
        "onSelect": undefined,
        "onSelectAll": undefined,
        "selected": Array [],
        "showOnlySelected": false,
        "unselectable": Array [],
      }
    }
    striped={false}
    tableBodyClass={null}
    tableContainerClass={null}
    tableHeaderClass={null}
    trClassName="submissionHover"
  >
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="_id"
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={true}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={true}
      searchable={true}
      sortIndicator={true}
      width={null}
    />
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="updatedAt"
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={true}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width={null}
    />
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="primaryInsuredName"
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width="100px"
    >
      Primary Named Insured
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="quoteStatus"
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width="35px"
    >
      Was 
      <br />
      Submission 
      <br />
      Priced?
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="quotedPremium"
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width="40px"
    >
      Quoted 
      <br />
      Premium
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="totalCost"
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width="40px"
    >
      Total 
      <br />
      Cost
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="totalPremium"
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width="40px"
    >
      Total 
      <br />
      Premium
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="type"
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width="20px"
    >
      Type
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="dateCreated"
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width="55px"
    >
      Date 
      <br />
      Created
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      dataAlign="left"
      dataField="dateUpdated"
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      searchable={true}
      sortIndicator={true}
      width="55px"
    >
      Date 
      <br />
      Updated
    </TableHeaderColumn>
  </BootstrapTable>
</div>
`;
